lib/spack/spack/cmd/debug.py

```diff
@@ -25,48 +25,65 @@ def setup_parser(subparser: argparse.ArgumentParser) -> None:
     sp.add_parser("report", help="print information useful for bug reports")
 
 
-def _get_builtin_repo_commit() -> Optional[str]:
+def _format_repo_info(source, commit):
+    if ".git" in source:
```

Comment:
Personal style: I would do `source.endswith(".git")` here and then do `f"{source[:-4]}/commit/{commit}"`